lib/cartopy/mpl/contour.py

```diff
@@ -52,7 +52,11 @@ def clabel(self, *args, **kwargs):
             # list in-place (as the contour label code does in mpl).
             paths = col.get_paths()
 
-            data_t = self.ax.transData
+            # The ax attribute is deprecated in MPL 3.3 in favor of
+            # axes. So, here we test if axes is present and fall back
+            # on the old self.ax to support MPL versions less than 3.3
+            data_t = getattr(self, "axes", self.ax).transData
```

Comment:
Won't this still warn about `self.ax`, since parameters are evaluated before the function call?

Reply:
Interesting, I guess I didn't think that self.ax would be called before passing it in to a function. However, that appears not to be the case here. The warnings disappear for me in the test suite with this implementation.

My thought is that this would short circuit if "axes" is present in self, just like an if statement.

I was just trying to shorten up:
```python
if hasattr(self, "axes"):
    data_t = self.axes.transData
else:
    data_t = self.ax.transData
```
I can change it to be a little more explicit if desired.

Reply:
Strange, it should definitely access it:
```
In [2]: class Foo: 
   ...:     @property 
   ...:     def bar(self): 
   ...:         print('accessing bar') 
   ...:     @property 
   ...:     def baz(self): 
   ...:         print('accessing baz') 
   ...:                                                                                                                                               

In [3]: f = Foo()                                                                                                                                     

In [4]: getattr(f, 'baz', f.bar)                                                                                                                      
accessing bar
accessing baz
```

Reply:
I must have had something messed up in my environment then... I have no idea why the warnings weren't being spit out.

I just updated the code to the if/else statement I put above to just be more explicit. Good catch!